### PR TITLE
Tweak build status a bit

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Make track healthy/exemplar when not having learning mode
- Use track-specific reputation for build status
- Show number of mentoring requests

### CODE REVIEW

File: app/commands/track/update_build_status.rb [hunk 1]
```diff
@@ -38,22 +38,16 @@ def component_health_statuses = [
...
-      track_id: track.id
-    ).select(:user_id)
-    volunteers = User.with_attached_avatar.where(id: volunteer_user_ids).order(reputation: :desc).take(12)
+    volunteers = AssembleContributors.({ track_slug: track.slug, page: 1 })
 
     {
-      num_volunteers: volunteer_user_ids.distinct.count,
-      users: SerializeAuthorOrContributors.(volunteers)
+      num_volunteers: volunteers.dig(:meta, :total_count),
+      users: volunteers[:results].take(NUM_VOLUNTEERS)
```
Comment (iHiD) [inline app/commands/track/update_build_status.rb]: While I like this idea, I don't think you want to do it. That AssembleContibutors thing is really slow due to the contextual data.

Instead, I'd go with your existing code, but replace the `where...select` with:
```
    User::ReputationPeriod::Search.(track_id: track.id)
```

which I think does what you want.

Comment (ErikSchierboom) [reply app/commands/track/update_build_status.rb]: But then I don't have the right, per-track reputation for volunteers.

Comment (iHiD) [reply app/commands/track/update_build_status.rb]: I think this is the going to be the most efficient way:

```
users = User::ReputationPeriod::Search.(track_id: track.id)
track_rep_mapping = User::ReputationToken.where(track_id: track.id).where(user: users).group(:user_id).sum(:value)
```

File: app/commands/track/update_build_status.rb [hunk 2]
```diff
@@ -38,22 +38,25 @@ def component_health_statuses = [
...
+    track_volunteers = User::ReputationPeriod::Search.(track_id: track.id)
+    num_volunteers = track_volunteers.total_count
+    top_track_volunteers = track_volunteers.take(NUM_VOLUNTEERS)
+
+    contextual_data = User::ReputationToken.
+      where(track:).
+      where(user: top_track_volunteers).
+      group(:user_id).
+      sum(:value).
+      transform_values { |reputation| { activity: '', reputation: } }
```
Comment (ErikSchierboom) [inline app/commands/track/update_build_status.rb]: Let me know what you think of this new implementation

File: app/commands/track/update_build_status.rb (hunk 1 above)
Comment (ErikSchierboom) [reply app/commands/track/update_build_status.rb]: I've changed the implementation. Let me know what you think

File: app/commands/track/update_build_status.rb [hunk 3]
```diff
@@ -220,12 +225,20 @@ def num_concepts_target
   end
 
   memoize
-  def taught_concepts = Concept.where(id: concept_taught_exercise.keys).to_a
+  def taught_concepts
+    Concept.where(id: concept_taught_exercise.keys).
```
Comment (iHiD) [inline app/commands/track/update_build_status.rb]:
```suggestion
    Concept.where(id: taught_concepts_ids).
```

**Review by iHiD — APPROVED** [review]

**Pull request merged**